src/gui/properties/peerlistsortmodel.h

```diff
@@ -31,29 +31,61 @@
 #ifndef PEERLISTSORTMODEL_H
 #define PEERLISTSORTMODEL_H
 
-#include <QStringList>
 #include <QSortFilterProxyModel>
+
 #include "peerlistdelegate.h"
 
-class PeerListSortModel : public QSortFilterProxyModel {
-  Q_OBJECT
+class PeerListSortModel: public QSortFilterProxyModel
+{
+    Q_OBJECT
 
 public:
-  PeerListSortModel(QObject *parent = 0) : QSortFilterProxyModel(parent) {}
+    PeerListSortModel(QObject *parent = 0)
+        : QSortFilterProxyModel(parent)
+    {
+    }
 
 protected:
-  bool lessThan(const QModelIndex &left, const QModelIndex &right) const {
-  switch (sortColumn()) {
-  case PeerListDelegate::IP:
-  case PeerListDelegate::CLIENT: {
-    QString vL = left.data().toString();
-    QString vR = right.data().toString();
-    return Utils::String::naturalCompareCaseSensitive(vL, vR);
-  }
-  };
-
-  return QSortFilterProxyModel::lessThan(left, right);
-  }
+    bool lessThan(const QModelIndex &left, const QModelIndex &right) const override
+    {
+        const bool isAscendingOrder = sortOrder() == Qt::AscendingOrder;
+
+        switch (sortColumn()) {
+        case PeerListDelegate::IP:
+        case PeerListDelegate::FLAGS: {
+            // put empty ones at the bottom of list
+            QString strL = left.data().toString();
+            QString strR = right.data().toString();
+            if (strL.isEmpty()) return !isAscendingOrder;
+            if (strR.isEmpty()) return isAscendingOrder;
+            return Utils::String::naturalCompareCaseSensitive(strL, strR);
+        }
+
+        case PeerListDelegate::CLIENT: {
+            // put empty ones at the bottom of list
+            QString strL = left.data().toString();
+            QString strR = right.data().toString();
+            if (strL.isEmpty()) return !isAscendingOrder;
+            if (strR.isEmpty()) return isAscendingOrder;
+            return (QString::compare(strL, strR, Qt::CaseInsensitive) < 0);
```

Comment:
The point is, the μ character seems to be differently encoded in the case of #5238.
i.e. the μ in  `μtorrent 3.4.3` & `μtorrent 3.2.1` are different!
Using natural sort seems to differentiate them, only using `QString::compare` could treat them as the same character.